roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -81,6 +90,19 @@ def run(self, tmp, task_vars):
             config_bytes = max(config.get(name, 0) for name in group_names) * 10**9
             recommended_bytes = config_bytes or recommended_bytes
 
+            # if an "upgrade" context is set, update the minimum disk requirement
+            # as this signifies an in-place upgrade - the node might have the
+            # required total disk space, but some of that space may already be
+            # in use by the existing OpenShift deployment.
+            context = get_var(task_vars, "r_openshift_health_checker_playbook_context", default="")
+            if context == "upgrade":
+                recommended_bytes = self.recommended_disk_upgrade_bytes.get(path, {})
```

Comment:
Could reuse the existing value of `recommended_bytes` instead of resetting it:

```python
recommended_bytes = self.recommended_disk_upgrade_bytes.get(path, recommended_bytes)
```